### PR TITLE
Feat: renameIndex method

### PR DESCRIPTION
**Pull request by Meldiron**
Adds `renameIndex` method

### CODE REVIEW

**Comment by Meldiron** [comment]
Should be safe to use without worrying about performance with big collections.

File: src/Database/Database.php [hunk 1]
```diff
@@ -631,6 +631,54 @@ public function renameAttribute(string $collection, string $id, string $name): b
...
+     */
+    public function renameIndex(string $collection, string $id, string $name): bool
+    {
+        $collection = $this->getCollection($collection);
+
+        $indexes = $collection->getAttribute('indexes', []);
+
+        $index = \array_search($id, \array_map(function($index) {
+            return $index['$id'];
+        }, $indexes));
```
Comment (TorstenDittmann) [inline src/Database/Database.php:651]: Instead of `array_map`, you can use `array_columns` - which will not do a iteration.

Comment (TorstenDittmann) [reply src/Database/Database.php:651]: and if you don't need the value of `$index`, you can use `in_array` - then you don't need to iterate once.

File: src/Database/Database.php [hunk 2]
```diff
@@ -631,6 +631,54 @@ public function renameAttribute(string $collection, string $id, string $name): b
...
+
+        if($index === false) {
+            throw new Exception('Index not found');
+        }
+
+        $indexNew = \array_search($name, \array_map(function($index) {
+            return $index['$id'];
+        }, $indexes));
```
Comment (TorstenDittmann) [inline src/Database/Database.php:659]: same comment as above.

File: src/Database/Database.php (hunk 1 above)
Comment (Meldiron) [reply src/Database/Database.php:651]: Done both, in both cases.

File: src/Database/Database.php (hunk 2 above)
Comment (Meldiron) [reply src/Database/Database.php:659]: Done both, in both cases.

**Review by TorstenDittmann — APPROVED** [review]

File: src/Database/Database.php [hunk 3]
```diff
@@ -631,6 +631,54 @@ public function renameAttribute(string $collection, string $id, string $name): b
...
+     */
+    public function renameIndex(string $collection, string $old, string $new): bool
+    {
+        $collection = $this->getCollection($collection);
+
+        $indexes = $collection->getAttribute('indexes', []);
+
+        $index = \in_array($old, \array_map(function($index) {
+            return $index['$id'];
+        }, $indexes));
```
Comment (TorstenDittmann) [inline src/Database/Database.php:651]: I think arrow functions are less noisy 👀

Comment (Meldiron) [reply src/Database/Database.php]: Updated.

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**